packages/gateway/src/handlers/address.ts

```diff
@@ -10,35 +10,64 @@ interface WriteRepository {
 export function withSetAddr(repo: WriteRepository): ccip.HandlerDescription {
   return {
     type: 'setAddr',
-    func: async (args: ethers.utils.Result) => {
+    func: async (args) => {
       const params: SetAddressProps = {
         node: args.node,
         coin: args.coin,
         addr: args.addr,
       }
-      if (!params.coin) params.coin = 60 // default: ether
+      if (params.coin === undefined) params.coin = 60 // default: ether
```

Comment:
Why does the conditional changed? Also, it is possible to receive zero or something that is not a number? If so, it needs to be handle.

Reply:
it isn't possible to be anything other than a number due to the input decoding on the ccip-read-server implementation.

the condition was changed to support 0 as a valid value